reana_job_controller/slurmcern_job_manager.py

```diff
@@ -0,0 +1,168 @@
+                 workflow_uuid=None, workflow_workspace=None,
+                 cvmfs_mounts='false', shared_file_system=False,
+                 job_name=None, kerberos=False):
+        """Instanciate Slurm job manager.
+
+        :param docker_img: Docker image.
+        :type docker_img: str
+        :param cmd: Command to execute.
+        :type cmd: list
+        :param env_vars: Environment variables.
+        :type env_vars: dict
+        :param job_id: Unique job id.
+        :type job_id: str
+        :param workflow_id: Unique workflow id.
+        :type workflow_id: str
+        :param workflow_workspace: Workflow workspace path.
+        :type workflow_workspace: str
+        :param cvmfs_mounts: list of CVMFS mounts as a string.
+        :type cvmfs_mounts: str
+        :param shared_file_system: if shared file system is available.
+        :type shared_file_system: bool
+        :param job_name: Name of the job
+        :type job_name: str
+        """
+        super(SlurmJobManagerCERN, self).__init__(
+            docker_img=docker_img, cmd=cmd,
+            env_vars=env_vars, job_id=job_id,
+            workflow_uuid=workflow_uuid,
+            job_name=job_name)
+        self.compute_backend = "Slurm"
+        self.workflow_workspace = workflow_workspace
+        self.cvmfs_mounts = cvmfs_mounts
+        self.shared_file_system = shared_file_system
+
+    def _transfer_inputs(self):
+        """Transfer inputs to SLURM submit node."""
+        stdin, stdout, stderr = \
+            self.slurm_connection.ssh_client.exec_command('cd /data && pwd')
+        self.slurm_home_path = \
+            stdout.read().decode("utf-8").rstrip()
+        SlurmJobManagerCERN.SLURM_WORKSAPCE_PATH = os.path.join(
+            self.slurm_home_path,
+            self.workflow_workspace[1:])
+        SlurmJobManagerCERN.REANA_WORKSPACE_PATH = self.workflow_workspace
+        self.slurm_connection.ssh_client.exec_command(
+            'cd {} && mkdir -p {}'.format(
+                self.slurm_home_path,
+                self.workflow_workspace[1:])
+        )
+        sftp = self.slurm_connection.ssh_client.open_sftp()
+        os.chdir(self.workflow_workspace)
+        for walker in os.walk(self.workflow_workspace):
+            try:
+                sftp.mkdir(
+                    os.path.join(
+                        self.slurm_home_path,
+                        walker[0][1:])
+                )
+            except Exception:
+                pass
+            for file in walker[2]:
+                sftp.put(
+                    os.path.join(walker[0], file),
+                    os.path.join(self.slurm_home_path, walker[0][1:], file)
+                )
+
+    @JobManager.execution_hook
+    def execute(self):
+        """Execute / submit a job with Slurm."""
+        self.slurm_connection = SSHClient(
+            hostname=current_app.config['SLURM_CLUSTER_HOSTNAME'],
+            username=os.environ.get('SLURMCERN_USER'),
```

Comment:
If https://github.com/reanahub/reana-job-controller/pull/199/files#r337360904 makes sense, we can rename these two variables as well i.e. `SLURM_CLUSTER_HEADNODE_SSH_USER` and `SLURM_CLUSTER_HEADNODE_SSH_PASSWORD`.

Reply:
What about removing username/password variables altogether and using Kerberos? @roksys Have you tried paramiko with Kerberos on CERN network? It would be nice to use the same mechanism as for HTCondor, and it would be nice to not require add password as a clear text secret...

```
$ pip install "paramiko[gssapi]"

client.connect(..., gss_auth=True)
```